docs/concepts/projects/search.md

```diff
@@ -18,6 +18,27 @@ However, it may be preferable to have a stronger governance framework for ML ass
 
 ## Lineage
 
-Hopsworks tracks the lineage (or provenance) of ML assets automatically for you. You can see what features are used in which feature view or training dataset. You can see what training dataset was used to train a given model. For assets that are managed outside of Hopsworks, there is support for the explicit definition of lineage dependencies.
+Hopsworks tracks lineage in two separate ways:
+* implicit
+* explicit
+
+Implicit provenance means that the user doesn't have to change his code and the system will determine links between different artifacts based on their usage(read/write) within the context of the same application. 
+
+Explicit provenance has the user, explicitly marking the links between artifacts by performing additional method calls within their code.
+
+Both of the approache have pros and cons. Implicit provenance may add links where there might be none, because maybe the mechanisms are not fine grained enough. For example, you read two feature groups and write two feature groups. 
+Depending on how fine grained the system is and where in the stack the tracking is being done, there might be links added between both read feature groups and both written feature groups, so four links, even though maybe there should only be two links. Perfect implicit provenance would have all components track all interactions and report them in a way that can be easily combined. If this is not possible, you might want to modify some core components, such as the file system and resource manager and collect imperfect implicit provenance from these, such as we do at Hopsworks. 
+Explicit provenance, however relies on the user remembering to mark each link and to delete the call when the link is removed at a later time. 
+
+The aim, however, it to combine both mechanism and try to get a view as good as possible of the actions happening in the system. The implicit provenance can act as a sort of a reminder to the user abut their explicit calls. For example, you have marked feature group X as a source of a link, but you no longer read it in the code, but we see in previous version of the code you did read it -  did you maybe forgot to remove the explicit provenance call.
+
+Hopsworks tracks the lineage (or provenance) of ML assets:
+* feature groups:
+	- external feature groups
+    - cached feature groups
+    - streaming feature groups
```

Comment:
We don't differentiate anywhere in the documentation between cached and streaming feature groups. Streaming is just a way of writing to Hopsworks. The fact that we have an internal representation of a feature group with streaming APIs should not be exposed to the user.

just say that it tracks feature groups.